src/main/java/com/smartsheet/wiremock/extensions/RequestDiff.java

```diff
@@ -0,0 +1,223 @@
+	private static final String SCENARIO_URL_PATH_FIELD = "urlPath";
+	private static final String SCENARIO_METHOD_FIELD = "method";
+	private static final String SCENARIO_HEADERS_FIELD = "headers";
+
+	static String getDiff(Request request, JsonNode scenario) {
+		return String.format("%s %s %s %s %s",
+				diffHeaders(request, scenario),
+				diffUrl(request, scenario),
+				diffMethod(request, scenario),
+				diffQueryParams(request, scenario),
+				diffBody(request, scenario)).trim();
+	}
+
+	private static String diffBody(Request request, JsonNode scenario) {
+
+		if(request.getMethod().toString().toUpperCase() == "GET"){
+			return "";
+		}
+
+		JsonNode scenarioBody = scenario.get(SCENARIO_REQUEST_FIELD).get(SCENARIO_BODY_FIELD);
+		if( scenarioBody == null){
+			return "Test scenario's request body was not defined or failed to parse.";
+		}
+
+		String result = "";
+
+		String scenarioBodyString = scenarioBody.toString();
+		String requestBodyString = request.getBodyAsString();
+
+		try {
+			assertJsonEquals(scenarioBodyString, requestBodyString);
+		}
+		catch(AssertionError e) {
+			result = "Body differs from expectation - " + e.getMessage();
+		}
+
+		return result;
+	}
+
+	private static String diffUrl(Request request, JsonNode scenario) {
+		JsonNode scenarioURL = scenario.get(SCENARIO_REQUEST_FIELD).get(SCENARIO_URL_PATH_FIELD);
+		if(scenarioURL == null){
+			return "Test scenario's request URL path was not defined or failed to parse.";
+		}
+		String scenarioURLString = scenarioURL.textValue();
+		String requestURLString = request.getUrl().split("\\?")[0];
+
+		scenarioURLString = removeTrailingSlash(scenarioURLString);
+		requestURLString = removeTrailingSlash(requestURLString);
+
+		if (requestURLString.equals(scenarioURLString)) {
+			return "";
+		}
+
+		return formatAssert("URL Match", scenarioURLString, requestURLString);
+	}
+
+	private static String removeTrailingSlash(String str) {
+		if (str.charAt(str.length() - 1) == '/') {
+			return str.substring(0, str.length() - 1);
+		}
+
+		return str;
+	}
+
+	private static String diffMethod(Request request, JsonNode scenario) {
+		JsonNode scenarioMethod = scenario.get(SCENARIO_REQUEST_FIELD).get(SCENARIO_METHOD_FIELD);
+		if(scenarioMethod == null){
+			return "Test scenario's request method was not defined or failed to parse.";
+		}
+
+		String scenarioMethodString = scenarioMethod.textValue().toUpperCase();
+		String requestMethodString = request.getMethod().toString().toUpperCase();
+
+		if (requestMethodString.equals(scenarioMethodString)) {
+			return "";
+		}
+
+		return formatAssert("HTTP Method Match", scenarioMethodString, requestMethodString);
+	}
+
+	private static String diffQueryParams(Request request, JsonNode scenario) {
+		JsonNode scenarioQueryParams = scenario.get(SCENARIO_REQUEST_FIELD).get(SCENARIO_QUERY_PARAMETERS_FIELD);
+		if(scenarioQueryParams == null) {
+			ObjectMapper mapper = new ObjectMapper();
+			scenarioQueryParams = mapper.createObjectNode();
```

Comment:
I changed this to allow us to catch cases where the scenario specifies no query params, but the request supplies them

Reply:
Great catch. I hadn't looked at this logic too much. Only made sure to include the compare for that scenario in the diffUnexpectedQueryParams().